front/src/applications/operationalStudies/components/MacroEditor/ngeToOsrd.ts

```
@@ -123,9 +138,14 @@ const createTrainSchedulePayload = (
     return [originPathItem];
   });
 
+  const trainrunLabels = trainrun.labelIds.map(
+    (labelId) => labels.find((label) => label.id === labelId)?.label
```

There should always be a match here, no?

---

No, a train doesn't always have a label.

---

But all label IDs in train runs must always have an entry in the dto labels array?

---

Yes, there will always be a match.
But i still have a TS error if I don't put "?"

---

I'd suggest using `!` then, that way we'll get an error (which we can debug) in case this assumption ends up being broken, instead of silently ignoring the issue.

---

Ok, this solution also seems the best thank you for this catch.

---

Done